src/nwm_routing/src/nwm_routing/__main__.py

```diff
@@ -634,6 +627,76 @@ def main_v02(argv):
         if debuglevel <= -1:
             print(flowveldepth)
 
+    # directory containing WRF Hydro restart files
+    wrf_hydro_restart_dir = output_parameters.get(
+        "wrf_hydro_channel_restart_directory", None
+    )
+    if wrf_hydro_restart_dir:
+
+        wrf_hydro_channel_restart_new_extension = output_parameters.get(
+            "wrf_hydro_channel_restart_new_extension", "TRTE"
+        )
+
+        # list of WRF Hydro restart files
+        wrf_hydro_restart_files = list(
+            Path(wrf_hydro_restart_dir).glob(
+                output_parameters["wrf_hydro_channel_restart_pattern_filter"]
+                + "[!"
+                + wrf_hydro_channel_restart_new_extension
+                + "]"
+            )
+        )
+
+        if len(wrf_hydro_restart_files) > 0:
+            qvd_columns = pd.MultiIndex.from_product(
+                [range(nts), ["q", "v", "d"]]
+            ).to_flat_index()
+
+            flowveldepth = pd.concat(
+                [pd.DataFrame(r[1], index=r[0], columns=qvd_columns) for r in results],
+                copy=False,
+            )
+            nhd_io.write_channel_restart_to_wrf_hydro(
+                flowveldepth,
+                wrf_hydro_restart_files,
+                restart_parameters.get("wrf_hydro_channel_restart_file"),
+                run_parameters.get("dt"),
+                run_parameters.get("nts"),
+                restart_parameters.get("wrf_hydro_channel_ID_crosswalk_file"),
+                restart_parameters.get(
+                    "wrf_hydro_channel_ID_crosswalk_file_field_name"
+                ),
+                wrf_hydro_channel_restart_new_extension,
+            )
+        else:
+            # print error and/or raise exception
+            print(
+                "WRF Hydro restart files not found - Aborting restart write sequence"
+            )
+            raise AssertionError
```

Comment:
Is there a reason not to use the `raise Assertion(message: str)` construct here instead of printing the message separately?

Reply:
No reason - I just didn't know how to correctly use `raise AssertionError` when I first designed this. Thanks for noticing this!

Reply:
@rcabell -- like this?
``` python
else:
    # print error and raise exception
    str = "WRF Hydro restart files not found - Aborting restart write sequence"
    raise AssertionError(str)
```

Reply:
Yes, that's the idea. You can pass the string in directly, too. By doing this you get two advantages (beyond simpler code): the error will print to stderr vs stdout, and if a outer `except` block handles the error, it can inspect the message and print it / log it / do whatever with it.